source/common/router/config_impl.cc

```diff
@@ -87,7 +87,8 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost, const Json:
       host_redirect_(route.getString("host_redirect", "")),
       path_redirect_(route.getString("path_redirect", "")), retry_policy_(route),
       rate_limit_policy_(route), shadow_policy_(route),
-      priority_(ConfigUtility::parsePriority(route)) {
+      priority_(ConfigUtility::parsePriority(route)),
+      opaque_config_(route.getObject("opaque_config")) {
```

Comment:
I'm not sure that is safe. The main JSON file gets blown away and your are going to point to nothing. This is just a weak pointer. Your test doesn't cover this because you keep the JSON parent alive. I would try switching your test to destroy the loader before you run the tests and it will probably crash.

With that said, I'm not crazy about returning raw JSON here. Can we potentially just do std::unordered_map<std::string, std::string>? Is that general enough for now?